### PR TITLE
Feature/prefect flow

### PR DESCRIPTION
**Pull request by shuchitach**
Added module for calling prefect flow

### CODE REVIEW

**Comment by shuchitach** [comment]
> Can you fix the test?

Yes, I need to get the prefect with hash added in requriements.txt

File: requirements.txt [hunk 1]
```diff
@@ -61,3 +61,6 @@ urllib3==1.26.6; python_version >= "2.7" and python_full_version < "3.0.0" or py
 uvicorn==0.15.0 \
     --hash=sha256:17f898c64c71a2640514d4089da2689e5db1ce5d4086c2d53699bf99513421c1 \
     --hash=sha256:d9a3c0dd1ca86728d3e235182683b4cf94cd53a867c288eaeca80ee781b2caff
+prefect==1.1.0 \
```
Comment (the-schwartz) [inline requirements.txt]: Ah we're using Poetry for our dependency management as per the Readme, so you'll need to add prefect to the pyproject.toml. You can install poetry first, `pip install poetry`, then add prefect with `poetry add prefect`

File: app/main.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 from fastapi import Depends, FastAPI, Header, HTTPException
 from app.liveness import liveness
 from app.workflow import wf_router
+from app.workflow import wf_service
```
Comment (the-schwartz) [inline app/main.py]: this shouldn't need to be imported here, can delete

File: app/workflow/wf_service.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                }
+            }
+        '''
+
+        id = client.graphql(flowId_by_name, variables={'name': name})
+        flow_run = client.create_flow_run(flow_id = id['data']['flow'][0]['id'])
+        return flow_run
+    
+    except:
+        # If flow name doesn't exist, return HTTP 404
```
Comment (the-schwartz) [inline app/workflow/wf_service.py]: comment isn't really needed as your code is self-commenting, you can remove

File: app/workflow/wf_service.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+from prefect import Client
+from fastapi import HTTPException
+
+# Replace with Prefect server host:port
+prefect_server = "http://20.25.85.193:4200/graphql"
```
Comment (the-schwartz) [inline app/workflow/wf_service.py]: externalizing this URI would be great - i don't have anything set up yet so if you'd like: https://www.dynaconf.com is a good way to go about it

File: app/workflow/wf_service.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+client = Client(
+    api_server = prefect_server
+)
+
+# Get the flow id of latest version using flow name
+def prefect_flow(name):
+    try: 
+        flowId_by_name = '''
```
Comment (the-schwartz) [inline app/workflow/wf_service.py]: mixed camelCase with snake_case here, `flow_id_by_name` would be preferred

File: app/workflow/wf_service.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+prefect_server = "http://20.25.85.193:4200/graphql"
+
+client = Client(
+    api_server = prefect_server
+)
+
+# Get the flow id of latest version using flow name
+def prefect_flow(projectName, flowName):
+    try: 
+        projectID_by_name = '''
```
Comment (the-schwartz) [inline app/workflow/wf_service.py]: camel and snake case mixup here as well - should be `project_id_by_name`

File: app/workflow/wf_service.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+# Replace with Prefect server host:port
+prefect_server = "http://20.25.85.193:4200/graphql"
+
+client = Client(
+    api_server = prefect_server
+)
+
+# Get the flow id of latest version using flow name
+def prefect_flow(projectName, flowName):
```
Comment (the-schwartz) [inline app/workflow/wf_service.py]: both these args should be snake case as well - project_name, flow_name

File: app/workflow/wf_router.py [hunk 8]
```diff
@@ -17,3 +18,8 @@ async def start_wf(request: StartWorkflowRequest):
 @router.post("/topology", response_model=WorkflowResponse)
 async def start_topology_wf(request: Topology):
     return WorkflowResponse(correlation_id = '', namespace = request.namespace, status = "In Progress")
+
+@router.post("/{projectName}/{flowName}")
+async def prefect_run_flow(projectName, flowName):
```
Comment (the-schwartz) [inline app/workflow/wf_router.py]: snake_case plz

**Review by the-schwartz — APPROVED** [review]

**Pull request merged**